manager/state/raft/storage.go

```diff
@@ -274,25 +275,112 @@ func (n *Node) saveSnapshot(snapshot raftpb.Snapshot, keepOldSnapshots uint64) e
 	// This means that if the current snapshot doesn't appear in the
 	// directory for some strange reason, we won't delete anything, which
 	// is the safe behavior.
-	var (
-		afterCurSnapshot bool
-		removeErr        error
-	)
+	curSnapshotIdx := -1
+	var removeErr error
+
 	for i, snapFile := range snapshots {
-		if afterCurSnapshot {
-			if uint64(len(snapshots)-i) <= keepOldSnapshots {
-				return removeErr
-			}
-			err := os.Remove(filepath.Join(n.snapDir(), snapFile))
-			if err != nil && removeErr == nil {
-				removeErr = err
+		if curSnapshotIdx >= 0 && i > curSnapshotIdx {
```

Comment:
Can we construct `remainingSnapshots` within this loop? So that it can reduce disk I/O by avoiding the call of `ioutil.ReadDir`, (or it may not harm the performance because of cache? I don't know).

Also, it may reduce some redundant code.